common/src/main/java/com/wynntils/wc/utils/SpellManager.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wc.utils;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.mc.event.ClientTickEvent;
+import com.wynntils.mc.event.SubtitleSetTextEvent;
+import com.wynntils.wc.event.WorldStateEvent;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class SpellManager {
+    private static final Pattern SPELL_PATTERN = Pattern.compile(
+            "§([LR]|Right|Left)§-§([LR?]|Right|Left)§-§([LR?]|Right|Left)§".replace("§", "(?:§[0-9a-fklmnor])*"));
+    public static final SpellUnit[] NO_SPELL = new SpellUnit[0];
+    private static SpellUnit[] lastSpell = NO_SPELL;
+    private static int spellCountdown = 0;
+
+    private static SpellUnit[] getSpellFromString(String string) {
+        Matcher spellMatcher = SPELL_PATTERN.matcher(string);
+        if (!spellMatcher.matches()) return null;
+
+        int size = 1;
+        for (; size < 3; ++size) {
+            if (spellMatcher.group(size + 1).equals("?")) break;
+        }
+
+        SpellUnit[] spell = new SpellUnit[size];
+        for (int i = 0; i < size; ++i) {
+            spell[i] = spellMatcher.group(i + 1).charAt(0) == 'R' ? SpellUnit.RIGHT : SpellUnit.LEFT;
+        }
+
+        return spell;
+    }
+
+    public static void tryUpdateSpell(String text) {
```

Comment:
`tryUpdateSpell` is called from both `SubtitleSetTextEvent` and `ActionBarManager`, is this intentional?

Reply:
This was intentional, however I am not really happy with this being outside of SpellManager. This was to avoid putting another actionbar parsing thing outside of ActionBarManager.